packages/input-datepicker/src/LionInputDatepicker.js

```diff
@@ -265,6 +265,7 @@ export class LionInputDatepicker extends OverlayMixin(LionInputDate) {
       contentNode,
       invokerNode,
       elementToFocusAfterHide: invokerNode,
+      hidesOnOutsideClick: true,
```

Comment:
I believe this should be done on the `modalDialog` directly, it applies to all overlays that can be closed on ESC.